src/databricks/labs/ucx/azure/resources.py

```diff
@@ -140,16 +181,64 @@ def post(self, path: str, body: dict | None = None):
             return self.api_client.do("POST", path, query, headers, body)
         return self.api_client.do("POST", path, query, headers)
 
-    def delete(self, path: str):
+    def delete(self, path: str, api_version: str | None = None):
         # this method is added only to be used in int test to delete the application once tests pass
         headers = {"Content-Type": "application/json"}
         query: dict[str, str] = {}
+        if api_version is not None:
+            query = {"api-version": api_version}
         return self.api_client.do("DELETE", path, query, headers)
 
     def token(self):
         return self._token_source.token()
 
 
+class AccessConnectorClient:
```

Comment:
For simplicity, shouldn't we just add all these methods to AzureResources? That's where we have all other azure resource management. We're not building an SDK library for Azure, we're using the apis we need